src/types/IRepository.ts

```diff
@@ -3,6 +3,7 @@ import { Repository } from "typeorm";
 type IRepository<T> = Repository<T> & {
   findById?(id: number, options?:any):Promise<T>;
   findByEmail?(email:string):Promise<T>;
+  findByAuth0_id?(auth0_id:string):Promise<T>;
```

Comment:
camelCase! 

We are starting to have too many user-specific methods in this interface which is going to be shared across entities. We should create another type that extends this one such as `IUserRepository` and adds `findByEmail` and `findByAuth0Id`.